crates/bevy_input/src/touch.rs

```diff
@@ -203,12 +203,18 @@ impl Touches {
                 }
             }
             TouchPhase::Ended => {
-                self.just_released.insert(event.id, event.into());
-                self.pressed.remove_entry(&event.id);
+                if let Some((_, v)) = self.pressed.remove_entry(&event.id) {
+                    self.just_released.insert(event.id, v);
```

Comment:
Shouldn't we update the touch phase before inserting? (Same on canceled)

Reply:
The touch info is from `self.pressed` or event, so I think this is fine.
I can change this if you prefer to update it first.

Reply:
Actually, the touch info should come from `self.pressed`.
If not found in `self.pressed`, there may be some error occurred, set it to default just to get rid of panic.

Reply:
> The touch info is from `self.pressed` or event, so I think this is fine.

You mean it already reflects the `Ended`/`Canceled` phase? I haven't used touches in Bevy before but I assume the event returned from `just_released` would be one of those phases.

Reply:
The info is updated from phase [TouchPhase::Moved](https://github.com/bevyengine/bevy/blob/main/crates/bevy_input/src/touch.rs#L196-L204)

Reply:
Oh interesting. See I'd think that one should also update the phase to `Moved`. 🤔

But maybe I'm just mistaken in how the touch system is supposed to work.